src/Core/Transform/UnitVector.cpp

```diff
@@ -303,4 +303,8 @@ namespace obe::Transform
     {
         return std::sqrt(std::pow(x - vec.x, 2) + std::pow(y - vec.y, 2));
     }
+    double UnitVector::magnitude() const
+    {
+        return std::sqrt(std::pow(x, 2) + std::pow(y, 2));
```

Comment:
Maybe, it should use a standardized unit ?

Reply:
Always returning the result in SceneUnits ?

Reply:
Yup for example, we can merge as-is for now since I'll have to do some refactor on UnitVector for v0.5 anyway, I'll modify the code later !